src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.service.impl.BulldozerProducerImpl;
+import core.mate.academy.service.impl.ExcavatorProducerImpl;
+import core.mate.academy.service.impl.TruckProducerImpl;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
```

Comment:
The class implementation should not begin with an empty line. Please remove the excess empty lines before the class declaration.